Rest/Controller/ContentController.php

```diff
@@ -0,0 +1,203 @@
+
+/**
+ * This file is part of the EzSystemsRecommendationBundle package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\RecommendationBundle\Rest\Controller;
+
+use eZ\Bundle\EzPublishCoreBundle\Imagine\AliasGenerator as ImageVariationService;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\API\Repository\Values\ContentType\ContentType;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\REST\Server\Controller as BaseController;
+use EzSystems\RecommendationBundle\Rest\Values\ContentData as ContentDataValue;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface as UrlGenerator;
+
+/**
+ * Recommendation REST Content controller.
+ */
+class ContentController extends BaseController
+{
+    /** @var ConfigResolverInterface */
+    protected $configResolver;
+
+    /** @var ImageVariationService */
+    protected $imageVariationService;
+
+    /** @var UrlGenerator */
+    protected $generator;
+
+    /**
+     * @param ConfigResolverInterface $configResolver
+     * @param ImageVariationService   $imageVariationService
+     * @param UrlGenerator            $generator
+     */
+    public function __construct(ConfigResolverInterface $configResolver, ImageVariationService $imageVariationService, UrlGenerator $generator)
+    {
+        $this->configResolver = $configResolver;
+        $this->imageVariationService = $imageVariationService;
+        $this->generator = $generator;
+    }
+
+    /**
+     * Prepares content for ContentDataValue class.
+     *
+     * @param $contentId
+     *
+     * @return ContentDataValue
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException     if the content, version with the given id and languages or content type does not exist
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException If the user has no access to read content and in case of un-published content: read versions
+     */
+    public function getContent($contentId)
+    {
+        $contentIds = explode(',', $contentId);
+        $content = $this->prepareContent($contentIds);
+
+        return new ContentDataValue($content);
+    }
+
+    /**
+     * Prepare content array.
+     *
+     * @param array $contentIds
+     *
+     * @return array
+     */
+    protected function prepareContent($contentIds)
+    {
+        $contentService = $this->repository->getContentService();
+        $locationService = $this->repository->getLocationService();
+        $contentTypeService = $this->repository->getContentTypeService();
+
+        $requestLanguage = $this->request->get('lang');
+        $requestedFields = $this->request->get('fields');
+
+        $content = array();
+
+        foreach ($contentIds as $contentId) {
+            $contentValue = $contentService->loadContent((int) $contentId, [$requestLanguage]);
```

Comment:
I'm not sure if int casting is a good direction here. According to https://github.com/ezsystems/EzSystemsRecommendationBundle/pull/14#discussion_r33783588 this might change in the future. Maybe @bdunogier can give us more details here.